core/server/master/src/main/java/alluxio/master/file/replication/ReplicationChecker.java

```diff
@@ -252,9 +228,11 @@ private void checkMisreplicated(Set<Long> inodes, ReplicationHandler handler)
           for (Map.Entry<String, String> entry
               : findMisplacedBlock(file, blockInfo).entrySet()) {
             try {
-              final long jobId =
-                  handler.migrate(inodePath.getUri(), blockId, entry.getKey(), entry.getValue());
-              mActiveJobToInodeID.put(jobId, inodeId);
+              if (mRequestBatchCount >= mMaxSingleBatch) {
+                return;
+              }
+              handler.migrate(inodePath.getUri(), blockId, entry.getKey(), entry.getValue());
+              mRequestBatchCount++;
```

Comment:
is it possible to have race here?

Reply:
Replication checker is currently single threaded. Will need to consider race when we multi thread it.